core/src/main/scala/com/pingcap/tispark/TiBatchWrite.scala

```diff
@@ -131,56 +153,14 @@ class TiBatchWrite(@transient val df: DataFrame,
     // TODO: lock table
     // pending: https://internal.pingcap.net/jira/browse/TIDB-1628
 
-    // TODO: if this write is update, TiDB reuses row_id. We need adopt this behavior.
-    // TODO: 1. fill column with its default value if it is null
-    //       2. refactor the writing process
-
-    // when primary is handle, it does not require allocate ids for each row.
-    val offset = if (!tiTableInfo.isPkHandle) {
-      val rowIDAllocator = RowIDAllocator.create(
-        tiDBInfo.getId,
-        tiTableInfo.getId,
-        tiSession.getCatalog,
-        tiTableInfo.isAutoIncColUnsigned,
-        df.count
-      )
-      rowIDAllocator.getStart
-    } else {
-      0
-    }
-
-    // i + start will be handle id if primary key is not handle
-    val tiKVRowRDD = df.rdd.zipWithIndex.map {
-      case (row, i) =>
-        sparkRow2TiKVRow(i + offset, row)
-    }
-
-    // deduplicate
-    val deduplicateRDD = deduplicate(tiKVRowRDD)
-
-    // encode TiROW
-    val encodedTiRowRDD = deduplicateRDD.map {
-      case (key, tiRow) => (key, tiRow, encodeTiRow(tiRow))
-    }
+    val tiRowRdd = df.rdd.map(row => sparkRow2TiKVRow(row))
 
-    // region pre-split
-    if (enableRegionPreSplit) {
-      logger.info("region pre split is enabled.")
-      val sampleRDD =
-        encodedTiRowRDD.sample(withReplacement = false, fraction = options.sampleFraction)
-      val (dataSize, sampleRDDCount) = estimateDataSize(sampleRDD, options)
+    val deduplicatedTiRowRdd = deduplicateIfNecessary(tiRowRdd, tiTableInfo.isPkHandle)
 
-      tiContext.tiSession.splitRegionAndScatter(
-        calculateSplitKeys(
-          encodedTiRowRDD,
-          dataSize,
-          sampleRDDCount,
-          tiTableInfo,
-          isUpdate = false,
-          regionSplitNumber
-        ).map(k => k.bytes).asJava
-      )
-    }
+    // TODO support insert ignore
+    val toBeCheckedRdd = getKeysNeedCheck(deduplicatedTiRowRdd)
+    val checkedConflictRdd = checkConflictWithInsertKeys(toBeCheckedRdd)
```

Comment:
`checkConflictWithInsertKeys` should return `Unit` instead of `RDD`

Reply:
I tried Uint, but checkConflictWithInsertKeys is skipped. That is why I chaned the return value from Uint to RDD.